lib/utils/parse-Approve.ts

```diff
@@ -0,0 +1,47 @@
+import { ERC20_ABI, ERC721_ABI, PERMIT2_ABI } from 'lib/abis';
+import { DecodedEventLog, Log, ParsedEvent } from 'lib/interfaces';
+import { decodeEventLog } from 'viem';
+
+export const parseLog = (log: Log): ParsedEvent | null => {
+  const abis = [ERC20_ABI, ERC721_ABI, PERMIT2_ABI];
+
+  for (const abi of abis) {
+    try {
+      const decodedLog = decodeEventLog({
+        abi: abi,
+        data: log.data,
+        topics: log.topics,
+      }) as DecodedEventLog;
+
+      const args = decodedLog.args;
+
+      if (decodedLog.name === 'Approval') {
```

Comment:
We also need to account for the ERC721 single NFT `Approval` here, which will have an extra argument for `tokenId`.